src/ringmesh/geomodel/builder/geomodel_builder_geology.cpp

```diff
@@ -216,8 +238,27 @@ namespace RINGMesh
         auto relationship_id = gmme_access.modifiable_parents()[id];
         auto& manager = geomodel_access_.modifiable_entity_type_manager()
                             .relationship_manager;
+        const auto old_parent_gmge = mesh_entity.parent_gmge( id );
         manager.set_parent_to_parent_child_relationship(
             relationship_id, parent_gmge );
+        if( update_parent )
+        {
+            // Add child in new parent relationships
+            GeoModelGeologicalEntityAccess< DIMENSION > parent_access{
```

Comment:
Maybe this bock can be in a different function to split.

Reply:
Done